### PR TITLE
chore(uve): Add Persona Selector to New Toolbar

### LINKED ISSUE #30736: FTM: New UVE Toolbar - Implement Persona selector #30735
### Parent Issue

https://github.com/dotCMS/core/issues/30598

### Task

Integrate the existing Persona Selector component into the new UVE Toolbar, maintaining its current design and functionality. This task involves adapting the `newWithEditorToolbar()` feature store to properly handle the Persona Selector state and interactions.

### Proposed Objective

Successfully integrate the Persona Selector into the new toolbar while ensuring identical behavior and functionality as in the old toolbar, across both traditional and headless pages.

### Acceptance Criteria

- Integrate the existing Persona Selector component into the new toolbar UI without visual modifications.
- Adapt the new toolbar store to handle Persona Selector state and actions:
  - Persona selection/switching
  - Current persona state
  - Persona list management
  - Any existing interactions with other toolbar components
- Ensure the component maintains exact feature parity with its implementation in the old toolbar:
  - Persona switching functionality
  - Persona list display
  - Current persona indication
  - Any existing keyboard navigation
- Verify the component works correctly in both traditional and headless pages.

### LINKED ISSUE #30863: UVE: Changes to specific persona are being overflowed to the default persona
### Parent Issue

TBD

### Problem Statement

When deleting content in a page for an specific persona, some changes are being saved for the default persona page.

Leading to unexpected modifications and errors in the content we are editing.

### Steps to Reproduce

https://github.com/user-attachments/assets/ae421959-6720-4fe5-a045-b99fa36e51a2

### Acceptance Criteria

- We should be able to edit content for an specific persona without messing with the default persona content

### dotCMS Version

trunk-latest

### Proposed Objective

User Experience

### Proposed Priority

Priority 1 - Show Stopper

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

- You understand the content edition flow in UVE 
- You understand the API Model is expected by the backend

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by zJaaal**
This pull request introduces several enhancements to the `dot-uve-toolbar` component, focusing on improving persona management and adding new services. The most significant changes include the integration of a persona selector component, updates to handle persona selection and despersonalization, and modifications to the state management for persona properties.

This PR also fixes a found issue, where every time we deleted content from a page for a specific persona, there were cases where the content was being save for the default persona, messing up with the page for the default persona.

## Before the fix
https://github.com/user-attachments/assets/ee61e3fe-9b0e-42b4-96ab-2045a8828573

## After the fix
https://github.com/user-attachments/assets/442279c4-1bb0-4e1e-8792-0b944d673c57

Enhancements to persona management:

* [`core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-uve-toolbar/dot-uve-toolbar.component.html`](diffhunk://#diff-9937556e73b051b878ba22ad1ce971a70019a617d7979b3e0bcc814801ad350bL41-R48): Replaced the static persona selector span with the `dot-edit-ema-persona-selector` component, adding event bindings for persona selection and despersonalization.

* [`core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-uve-toolbar/dot-uve-toolbar.component.ts`](diffhunk://#diff-217a9e619d6590c4f652e85353b9637ba5e464ddeb0424be35aef39bb8dceb30L2-R16): Added imports for `DotPersonalizeService` and `DotPersona`, included the `EditEmaPersonaSelectorComponent` in the component declarations, and introduced methods `onPersonaSelected` and `onDespersonalize` to handle persona-related actions. [[1]](diffhunk://#diff-217a9e619d6590c4f652e85353b9637ba5e464ddeb0424be35aef39bb8dceb30L2-R16) [[2]](diffhunk://#diff-217a9e619d6590c4f652e85353b9637ba5e464ddeb0424be35aef39bb8dceb30R27-R47) [[3]](diffhunk://#diff-217a9e619d6590c4f652e85353b9637ba5e464ddeb0424be35aef39bb8dceb30R60-R128)

State management updates:

* [`core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/models.ts`](diffhunk://#diff-a5d9d93bd678484a7d776d09685fcb638de6d969ac141586a1104ba5e744556bL131-L134): Removed the `personaSelector` property from `UVEToolbarProps` and created a new `PersonaSelectorProps` interface. [[1]](diffhunk://#diff-a5d9d93bd678484a7d776d09685fcb638de6d969ac141586a1104ba5e744556bL131-L134) [[2]](diffhunk://#diff-a5d9d93bd678484a7d776d09685fcb638de6d969ac141586a1104ba5e744556bR140-R144)

* [`core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/toolbar/withUVEToolbar.ts`](diffhunk://#diff-e6d3fb6319626fa85a4fc6894b57935843713366be593de6dd1dc5ed68bf6afcL25-R25): Updated the state management to include a computed `$personaSelector` property that returns the current persona selector properties. [[1]](diffhunk://#diff-e6d3fb6319626fa85a4fc6894b57935843713366be593de6dd1dc5ed68bf6afcL25-R25) [[2]](diffhunk://#diff-e6d3fb6319626fa85a4fc6894b57935843713366be593de6dd1dc5ed68bf6afcL112-R123)

* [`core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/toolbar/withUVEToolbar.spec.ts`](diffhunk://#diff-7a5de702ac1dc81304f4c31816f5c0363aa56141f8afa583a87856e7a0d8482dR11): Added a test case to verify the persona selector properties in the store. [[1]](diffhunk://#diff-7a5de702ac1dc81304f4c31816f5c0363aa56141f8afa583a87856e7a0d8482dR11) [[2]](diffhunk://#diff-7a5de702ac1dc81304f4c31816f5c0363aa56141f8afa583a87856e7a0d8482dL50) [[3]](diffhunk://#diff-7a5de702ac1dc81304f4c31816f5c0363aa56141f8afa583a87856e7a0d8482dR82-R88)

This PR fixes: #30736

### CODE REVIEW

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-uve-toolbar/dot-uve-toolbar.component.spec.ts [hunk 1]
```diff
@@ -244,6 +263,104 @@ describe('DotUveToolbarComponent', () => {
...
+                    accept: expect.any(Function),
+                    acceptLabel: 'dot.common.dialog.accept',
+                    header: 'editpage.personalization.confirm.header',
+                    message: 'editpage.personalization.confirm.message',
+                    reject: expect.any(Function),
+                    rejectLabel: 'dot.common.dialog.reject'
+                });
+            });
+
+            xit('should personalize - call service', () => {
```
Comment (KevinDavilaDotCMS) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-uve-toolbar/dot-uve-toolbar.component.spec.ts:337]: remove?

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-uve-toolbar/dot-uve-toolbar.component.spec.ts [hunk 2]
```diff
@@ -244,6 +263,104 @@ describe('DotUveToolbarComponent', () => {
...
+                expect(confirmationService.confirm).toHaveBeenCalledWith({
+                    accept: expect.any(Function),
+                    acceptLabel: 'dot.common.dialog.accept',
+                    header: 'editpage.personalization.delete.confirm.header',
+                    message: 'editpage.personalization.delete.confirm.message',
+                    rejectLabel: 'dot.common.dialog.reject'
+                });
+            });
+
+            xit('should dpersonalize - call service', () => {
```
Comment (KevinDavilaDotCMS) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-uve-toolbar/dot-uve-toolbar.component.spec.ts:360]: same here

**Review by KevinDavilaDotCMS — APPROVED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/toolbar/withUVEToolbar.ts [hunk 3]
```diff
@@ -109,14 +109,18 @@ export function withUVEToolbar() {
...
                 };
             }),
+            $personaSelector: computed<PersonaSelectorProps>(() => {
+                const pageAPIResponse = store.pageAPIResponse();
+
+                return {
+                    pageId: pageAPIResponse?.page.identifier,
+                    value: pageAPIResponse?.viewAs.persona ?? DEFAULT_PERSONA
+                };
+            }),
```
Comment (rjvelazco) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/toolbar/withUVEToolbar.ts:123]: Nicesu!

**Review by rjvelazco — APPROVED** [review]
LGTM!

File: core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/toolbar/withUVEToolbar.ts (hunk 3 above)
Comment (zJaaal) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/toolbar/withUVEToolbar.ts:123]: ariagtobosaimasu

**Pull request merged**